src/ClientMessage.ts

```diff
@@ -26,53 +24,65 @@ import Long = require('long');
 import {BitsUtil} from './BitsUtil';
 
 class ClientMessage {
-    buffer:  Buffer;
-    writeIndex:  number;
-    readIndex:  number;
 
-    constructor(payload_size:  number) {
-        this.buffer = new Buffer(payload_size);
-        this.buffer.fill(0, 0, BitsUtil.HEADER_SIZE + payload_size);
-        this.setDataOffset(BitsUtil.HEADER_SIZE);
-        this.setVersion(BitsUtil.VERSION);
-        this.writeIndex = 0;
-        this.readIndex = 0;
+    private buffer: Buffer;
+    private cursor: number = BitsUtil.HEADER_SIZE;
```

Comment:
As far as I see, there is another cursor in BufferWrapper class. One of them may be redundent